engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCache.scala

```diff
@@ -0,0 +1,60 @@
+package cromwell.engine.workflow.lifecycle.execution.callcaching
+
+import cromwell.backend.BackendJobExecutionActor.SucceededResponse
+import cromwell.core.{JobOutputs, WorkflowId}
+import cromwell.database.sql._
+import cromwell.database.sql.tables.CallCachingResultMetainfoEntry
+import cromwell.engine.workflow.lifecycle.execution.callcaching.EngineJobHashingActor.CallCacheHashes
+import cromwell.core.ExecutionIndex.IndexEnhancedIndex
+import wdl4s.values._
+import language.postfixOps
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+/**
+  * Given a database-layer CallCacheStore, this accesser can access the database with engine-friendly data types.
+  */
+class CallCache(database: CallCachingStore) {
+  def addToCache(workflowId: WorkflowId, callCacheHashes: CallCacheHashes, response: SucceededResponse)(implicit ec: ExecutionContext): Future[Unit] = {
+    val preparation = Try {
```

Comment:
When I still had `toResultSimpletons` as a `???`. I think I agree that it's now impossible